DM-49364: Switch LATISS/bestEffortIsr to IsrTaskLSST

mfisherlevine opened the pull request.

mfisherlevine committed:
Update to use new IsrTaskLSST

mfisherlevine committed:
Update to use new IsrTaskLSST

mfisherlevine committed:
Remove doBootstrap config option

mfisherlevine committed:
Remove doBootstrap config option

czwa commented on config/quickLookIsr.py:
```diff
@@ -1,6 +1,7 @@
 config.brighterFatterMaxIter = 2  # Uncomment this to remove test warning
+config.doDeferredCharge = False  # no calib for this yet
+config.doBootstrap = False
+config.doApplyGains = False
```
Unless you expect to never have a PTC, I would strongly suggest this be set to `doApplyGains=True`.  The new ISR defaults largely assume that everything has had gains applied so units are in electrons.  We also should have CTI calibrations in the next set, but that correction is far less likely to have visual impact on the images than `doApplyGains` (which should help remove amp offsets).

czwa reviewed (approved):
I'm not going to make the change request mandatory, but unless there's a compelling reason not to `doApplyGains=True`, I think that will improve things.

mfisherlevine committed:
Turn gain correction on

mfisherlevine committed:
Turn gain correction on

czwa committed:
CZW/RB: Working through IsrTaskLSST issues.

czwa committed:
CZW/RB: Working through IsrTaskLSST issues.

mfisherlevine committed:
Remove non-existent doWrite option

mfisherlevine committed:
Remove non-existent doWrite option

estevesjh committed:
Fix FootprintSet STDEV detection failure in runSourceDetection

Replace detectObjectsInExp with direct detection using Threshold.VALUE
and np.std() to avoid STDEVCLIP returning 0 for some guider images.
Add starMaskThreshold parameter to exclude bright pixels when computing
background standard deviation.

estevesjh committed:
Suppress pre-existing mypy union-attr errors in utils.py

estevesjh committed:
Add nPixMin parameter to runSourceDetection

estevesjh committed:
Add dither to coadd median to prevent STDEVCLIP=0

Add a uniform [0, 1) random dither per stamp before computing the
median coadd. This breaks integer quantization that causes the pixel
distribution to be so narrow that STDEVCLIP returns 0 on some guider
images. Fix suggested by Robert Lupton. (DM-54263)

estevesjh committed:
Use STDEVCLIP with sigma68 fallback for source detection

Replace the manual sigma68-only background estimation with STDEVCLIP
from lsst.afw.math, falling back to sigma68 if STDEVCLIP returns 0.
The dithered coadds (previous commit) prevent STDEVCLIP from failing,
but the fallback provides a safety net. Also adds a configurable
nPixMin parameter for footprint detection. (DM-54263)

estevesjh committed:
Replace type: ignore with assertion for altaz_begin

Address review feedback from Merlin: use explicit assertion
instead of mypy ignore comment.

(DM-54263)

czwa committed:
Update quickLook/bestEffortIsr to use IsrTaskLSST.

mfisherlevine committed:
Fix linting

czwa committed:
Remove unneeded import.

czwa committed:
Use IsrMockLSST data to avoid issues.

czwa committed:
Remove another unneeded import.

czwa committed:
Fix linting errors.

mfisherlevine committed:
Update collection comment and remove (inactive) breakpoint
